pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -380,6 +382,69 @@ pub(super) async fn download_index_part(
     }
 }
 
+pub(crate) async fn download_initdb_tar_zst(
+    conf: &'static PageServerConf,
+    storage: &GenericRemoteStorage,
+    tenant_id: &TenantId,
+    timeline_id: &TimelineId,
+) -> Result<(Utf8PathBuf, File), DownloadError> {
+    debug_assert_current_span_has_tenant_and_timeline_id();
+
+    let remote_path = remote_initdb_archive_path(tenant_id, timeline_id);
+
+    let timeline_path = conf.timelines_path(tenant_id);
+
+    if !timeline_path.exists() {
+        tokio::fs::create_dir_all(&timeline_path)
+            .await
+            .with_context(|| format!("timeline dir creation {timeline_path}"))
+            .map_err(DownloadError::Other)?;
+    }
+    let temp_path = timeline_path.join(format!("{INITDB_PATH}-{timeline_id}.{TEMP_FILE_SUFFIX}"));
+
+    let file = download_retry(
+        || async {
+            let mut file = OpenOptions::new()
+                .create(true)
+                .truncate(true)
```

Comment:
This code allows two concurrent tasks to incorrectly operate on the same inode.

Bettter to use `create_new` and handle the EEXIST error by unlinking the file and retrying.

I'd also move the temp file creation outside of the `download_retry`.`

Reply:
initdb runs prior to this and is already not race-free, because it uses the `basebackup-{timeline_id}.__temp` directory.